test/integration/upload/cli/test_upload.py

```diff
@@ -24,6 +25,19 @@ def setUp(self):
         self.area = self.mock_current_upload_area()
         self.test_files = ['LICENSE', 'README.rst']
 
+    def upload_command(self, args):
```

Comment:
Nit:
```
        paths = []
        if args.target_filename:
            paths.append(args.target_filename)
        for arg in args.upload_paths:
            if os.path.isdir(arg):
                paths.extend(itertools.chain.from_iterable((f for _, _, f in os.walk(arg))))
            else:
                paths.append(arg)
        for path in paths:
            self.add_upload_mock(self.area.uuid, path)
        return UploadCommand(args)
```

Could be shortened to:

```
        for arg in args.upload_paths + [args.target_filename or '']:
            if os.path.isdir(arg):
                [self.add_upload_mock(self.area.uuid, p) for l in [f for _, _, f in os.walk(arg)] for p in l]
            elif os.path.isfile(arg):
                self.add_upload_mock(self.area.uuid, arg)
        return UploadCommand(args)
```

Haha, you can ignore that one though if you like.

Though I'm wondering why add these mocked responses since you're just introducing a logging message condition?  Were the tests not running before without them?

Reply:
> Could be shortened to:

Updated with a similar version!

> Were the tests not running before without them?

Yes - the previous error suppression mechanism didn't just suppress logging of errors, it prevented errors from being raised at all, which caused tests that shouldn't have passed to pass silently. (All of those tests are fixed in this PR, though.)